django/applications/catmaid/control/graph2.py

```diff
@@ -276,20 +284,20 @@ def newSynapseCounts():
 
     # Create the edges of the graph
     edges = defaultdict(partial(defaultdict, newSynapseCounts)) # pre vs post vs count
-    for c in connectors.itervalues():
+    for c in six.itervalues(connectors):
         for pre in c[preID]:
             for post in c[postID]:
                 edges[pre[0]][post[0]][min(pre[1], post[1]) - 1] += 1
 
     return {'nodes': nodeIDs,
-            'edges': [(pre, post, count) for pre, edge in edges.iteritems() for post, count in edge.iteritems()],
+            'edges': [(pre, post, count) for pre, edge in six.iteritems(edges) for post, count in six.iteritems(edge)],
             'branch_nodes': branch_nodeIDs,
             'intraedges': intraedges}
 
 
 def populate_connectors(chunkIDs, chunks, cs, connectors):
     # Build up edges via the connectors
-    IDchunks = zip(chunkIDs, chunks)
+    IDchunks = list(izip(chunkIDs, chunks))
```

Comment:
There wouldn't be any need if we replace the inner loop's `in IDchunks` with the iterator based `zip` (which would be nicer indeed, i.e.: `in izip(chunkIDs, chunks)`). But with this variable in use, the iterator will only be run once for the inner loop. The second iteration of the outer loop will not re-run the iterator. For now I just tried to not change too much.